### PR TITLE
Generate local IP address

### PR DESCRIPTION
**Pull request by mfalesni**

### CODE REVIEW

File: fauxfactory/__init__.py [hunk 1]
```diff
@@ -522,6 +522,16 @@ def gen_ipaddr(ip3=False, ipv6=False):
...
 
 
+def gen_local_ipaddr():
+    """Generates a random local IP address (10.x.x.x subnet)
+
+    :returns: A random IP address from local 10.x.x.x subnet
+    :rtype: str
+    """
+    return "10.{}.{}.{}".format(
+        *[gen_integer(min_value=0, max_value=255) for i in range(3)])
```
Comment (Ichimonji10) [inline fauxfactory/__init__.py]: `s/i/_/`

File: fauxfactory/__init__.py [hunk 2]
```diff
@@ -522,6 +522,16 @@ def gen_ipaddr(ip3=False, ipv6=False):
     return _make_unicode(ipaddr)
 
 
+def gen_local_ipaddr():
+    """Generates a random local IP address (10.x.x.x subnet)
+
+    :returns: A random IP address from local 10.x.x.x subnet
+    :rtype: str
+    """
```
Comment (Ichimonji10) [inline fauxfactory/__init__.py]: Blank line between `:rtype:` and the closing quotes, please.

File: fauxfactory/__init__.py [hunk 3]
```diff
@@ -522,6 +522,16 @@ def gen_ipaddr(ip3=False, ipv6=False):
     return _make_unicode(ipaddr)
 
 
+def gen_local_ipaddr():
```
Comment (Ichimonji10) [inline fauxfactory/__init__.py]: This function should be added to `__all__`.

File: tests/test_ipaddress.py [hunk 4]
```diff
@@ -82,3 +82,19 @@ def test_gen_ipv6_2(self):
...
+        """
+        addr = gen_local_ipaddr()
+        self.assertTrue(
+            len(addr.split(".")) == 4, "IP address does not have 4 fields")
+        self.assertTrue(
+            addr.startswith("10."), "IP address does not start with 10")
+        fields = addr.split(".")
+        self.assertTrue(
+            all(int(field) < 256 and int(field) >= 0 for field in fields),
+            "IP address' field is bigger than 255!")
```
Comment (Ichimonji10) [inline tests/test_ipaddress.py]: Or the IP address field is less than zero.

**Comment by Ichimonji10** [comment]
I wonder if it's possible to fix up `gen_ipaddr` to handle use cases like this? One option is to add a `private` parameter to that function. Such an approach would be backwards-compatible. It could also be a bit nicer, as this call:

    gen_ipaddr()

...could return an address in the `10.0.0.0/24`, `172.16.0.0/20` or `192.168.0.0/16` range. This approach is also nice because it means we don't have to expand fauxfactory's namespace with a second function that is highly similar to an existing function.

**Comment by mfalesni** [comment]
@Ichimonji10 I would prefer that solution but I did not want to mess with the original code.

**Comment by omaciel** [comment]
>> I wonder if it's possible to fix up gen_ipaddr to handle use cases like this?

Thank you for your contribution @mfalesni I also think that we should modify the existing ``gen_ipaddr`` method to handle this case. Would you agree with doing it this way?

**Comment by mfalesni** [comment]
Yup.

**Comment by mfalesni** [comment]
Closign in favour of #75

**Pull request closed without merging**